### PR TITLE
Update Conda Recipe and `README.md`

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR bumps cudatoolkit version in conda recipe to 11.5 (and removes old versions). Updates `README.md` to reflect recent changes and verified build steps from bare metal.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,31 +41,36 @@ conda install -c conda-forge -c rapidsai-nightly cuspatial
...
+
+- gcc >= 7.5
+- cmake >= 3.23
+- miniconda
+
+### Fetch cuSpatial repository
+
+```shell
+export `CUSPATIAL_HOME=$(pwd)/cuspatial` && \
+git clone --recurse-submodules https://github.com/rapidsai/cuspatial.git $CUSPATIAL_HOME
```
Comment (harrism) [inline README.md:54]: Do we need the submodule option? Do we still have submodules? We got rid of them in other repos.

File: README.md [hunk 2]
```diff
@@ -39,31 +41,36 @@ conda install -c conda-forge -c rapidsai-nightly cuspatial
...
+Install all dependencies required from conda recipe:
 
-The rest of steps assume the environment variable `CUDF_HOME` points to the 
-root directory of your clone of the cuDF repo, and that the `cudf_dev` Anaconda
-environment created in step 3 is active.
+```shell
+conda env update --file conda/environments/cuspatial_dev_cuda11.5.yml
+```
 
 ### Clone, build and install cuSpatial
```
Comment (harrism) [inline README.md:64]:
```suggestion
### Build and install cuSpatial
```

**Review by harrism — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Review by thomcom — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (harrism) [reply README.md:54]: @isVoid can you answer this?

**Review by harrism — CHANGES_REQUESTED** [review]
Changing to request changes due to unanswered comments.

File: README.md (hunk 1 above)
Comment (isVoid) [reply README.md:54]: Somehow this PR fall through my tracks. No I don't think we have submodules in the repo.

**Review by harrism — APPROVED** [review]

**Comment by isVoid** [comment]
rerun tests

**Comment by isVoid** [comment]
@gpucibot merge

**Pull request merged**